// igdb-api-node/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "getTagNumber", {
  enumerable: true,
  get: function get() {
    return _tagNumber["default"];
  }
});
exports["default"] = void 0;

var _apicalypse = _interopRequireDefault(require("apicalypse"));

var _deepmerge = _interopRequireDefault(require("deepmerge"));

var _isPlainObject = _interopRequireDefault(require("is-plain-object"));

var _tagNumber = _interopRequireDefault(require("./tag-number"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

/**
 * Creates the IGDB API object, populated with methods for building a query.
 * @arg {string} [apiKey]
 * @returns {Object}
 */
var _default = function _default(apiKey) {
  var opts = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
  var key = apiKey || process.env.IGDB_API_KEY || global.IGDB_API_KEY;
  var defaultOptions = {
    method: 'POST',
    baseURL: 'https://api-v3.igdb.com',
    headers: {
      'user-key': key,
      accept: 'application/json'
    },
    responseType: 'json'
  };
  return (0, _apicalypse["default"])((0, _deepmerge["default"])(defaultOptions, opts, {
    isMergeableObject: _isPlainObject["default"]
  }));
};

exports["default"] = _default;
